classes/core/PKPRoutingProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace PKP\core;
+
+use Illuminate\Http\Request;
+use Illuminate\Routing\Router;
+use Illuminate\Routing\UrlGenerator;
+use Illuminate\Routing\RoutingServiceProvider;
+use Illuminate\Foundation\Http\Middleware\TrimStrings;
+use Illuminate\Foundation\Http\Middleware\ValidatePostSize;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull;
+use Illuminate\Contracts\Routing\UrlGenerator as UrlGeneratorContract;
+use Illuminate\Support\Facades\Response;
+use PKP\core\PKPContainer;
+use PKP\middleware\HasUser;
+use PKP\middleware\HasRoles;
+use PKP\middleware\HasContext;
+use PKP\middleware\AllowCrossOrigin;
+use PKP\middleware\ValidateCsrfToken;
+use Psr\Http\Message\ServerRequestInterface;
+use PKP\middleware\DecodeApiTokenWithValidation;
+use PKP\middleware\SetupContextBasedOnRequestUrl;
+
+
+class PKPRoutingProvider extends RoutingServiceProvider
+{
+    protected static $globalMiddleware = [
+        AllowCrossOrigin::class,
+        SetupContextBasedOnRequestUrl::class,
+        DecodeApiTokenWithValidation::class,
+        ValidateCsrfToken::class,
+        ValidatePostSize::class,
+        TrimStrings::class,
+        ConvertEmptyStringsToNull::class,
+    ];
+
+    /**
+     * The application's route middleware.
+     * These middleware can/should be assigned to specific routes or routes groups individually.
+     */
+    protected array $routeMiddleware = [
+        'has.roles'     => HasRoles::class,
+        'has.user'      => HasUser::class,
+        'has.context'   => HasContext::class,
+    ];
+
+    public static function getGlobalRouteMiddleware(): array
+    {
+        return self::$globalMiddleware;
+    }
+
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        parent::register();
+
+        $this->registerRouteMiddleware();
+        $this->registerRoutePatterns();
+        $this->registerResponseBindings();
+    }
+
+    /**
+     * Boot the service provider.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        Response::macro('withCSV', function (array $rows, array $columns, int $maxRows) { 
```

Comment:
I'm not sure @asmecher enjoys macros, I've tried to add one and it was rejected.
The `$rows`/`$columns` arguments could be iterators to give room for future optimizations.